cmd/tempo/app/modules.go

```diff
@@ -287,11 +287,12 @@ func (t *App) initQueryFrontend() (services.Service, error) {
 		httpGzipMiddleware(),
 	)
 
-	traceByIDHandler := middleware.Wrap(queryFrontend.TraceByID)
-	searchHandler := middleware.Wrap(queryFrontend.Search)
+	traceByIDHandler := middleware.Wrap(queryFrontend.TraceByIDHandler)
+	searchHandler := middleware.Wrap(queryFrontend.SearchHandler)
 
 	// register grpc server for queriers to connect to
 	frontend_v1pb.RegisterFrontendServer(t.Server.GRPC, t.frontend)
+	tempopb.RegisterStreamingQuerierServer(t.Server.GRPC, queryFrontend)
```

Comment:
Will this new server be traced for us?

Reply:
it should. tracing is initiated in weaveworks common and propagated correctly these changes (i think). if there are any issues I will patch them up.